pkg/controller/gitopsservice/gitopsservice_controller.go

```diff
@@ -264,7 +295,188 @@ func (r *ReconcileGitopsService) Reconcile(request reconcile.Request) (reconcile
+			reqLogger.Info("ArgoCD instance is not found or yet to be created", "Namespace", defaultArgoCDInstance.Namespace, "Name", defaultArgoCDInstance.Name)
+			return reconcile.Result{}, nil
+		}
+
+		if !instance.Spec.SSO {
+			reqLogger.Info("SSO is set to false, removing the argocd OIDC Configuration")
+			argoCD.Spec.OIDCConfig = ""
+			err = r.client.Update(context.TODO(), &argoCD)
+			if err != nil {
+				reqLogger.Error(err, "Error updating OIDC Config")
+				return reconcile.Result{}, err
+			}
+		} else {
+			reqLogger.Info("SSO is set to true, Setting up argocd OIDC Configuration")
+			// Create a new Keycloak Instance. Ignore if there is one already.
+			// Keycloak instance is created with default values out of the box. Admins or owners are allowed to modify the instance.
+			defaultKeycloakInstance := rhsso.NewKeycloakCR(gitopsIdentifier)
+			existingKeycloak := &keycloakv1alpha1.Keycloak{}
+
+			// Set GitopsService instance as the owner and controller
+			if err := controllerutil.SetControllerReference(instance, defaultKeycloakInstance, r.scheme); err != nil {
+				return reconcile.Result{}, err
+			}
+
+			err = r.client.Get(context.TODO(), types.NamespacedName{Name: defaultKeycloakInstance.Name, Namespace: defaultKeycloakInstance.Namespace}, existingKeycloak)
+			if err != nil {
+				reqLogger.Info("Creating a new Keycloak instance", "Namespace", defaultKeycloakInstance.Namespace, "Name", defaultKeycloakInstance.Name)
+				err = r.client.Create(context.TODO(), defaultKeycloakInstance)
+				if err != nil {
+					return reconcile.Result{}, err
+				}
+			}
+
+			// Create a new Keycloak Realm Instance. Ignore if there is one already.
+			// Keycloak Realm instance is created with default values out of the box. Admins or owners are allowed to modify the instance.
+			defaultKeycloakRealmInstance := rhsso.NewKeycloakRealmCR(gitopsIdentifier)
+			existingKeycloakRealm := &keycloakv1alpha1.KeycloakRealm{}
+
+			// Set GitopsService instance as the owner and controller
+			if err := controllerutil.SetControllerReference(instance, defaultKeycloakRealmInstance, r.scheme); err != nil {
+				return reconcile.Result{}, err
+			}
+
+			err = r.client.Get(context.TODO(), types.NamespacedName{Name: defaultKeycloakRealmInstance.Name, Namespace: defaultKeycloakRealmInstance.Namespace}, existingKeycloakRealm)
+			if err != nil && errors.IsNotFound(err) {
+				reqLogger.Info("Creating a new Keycloak Realm", "Namespace", defaultKeycloakRealmInstance.Namespace, "Name", defaultKeycloakRealmInstance.Name)
+				err = r.client.Create(context.TODO(), defaultKeycloakRealmInstance)
+				if err != nil {
+					return reconcile.Result{}, err
+				}
+			}
+
+			// Get Argocd Route URL and pass it to the Keycloak Client installation.
+			// Keycloak Client instance is created with default values out of the box. Admins or owners are allowed to modify the instance.
+			argoRoute := &routev1.Route{}
+			argoRouteRef := existingArgoRoute()
```

Comment:
Remind me, does the argocd controller create the route instantly, or does the route creation itself depend on something?

Reply:
it creates the route as soon as the instance gets created.